designdocs/wallettool-subcommands.adoc

```diff
@@ -0,0 +1,579 @@
+* This modularity reduces the likelihood of unrelated commands being affected when introducing changes and ensures that responsibilities are distributed cleanly across the system.
+
+- **Extensibility for Future Development**
+* Adding a new command becomes straightforward because it only requires creating a new subcommand and registering it, without altering existing logic.
+* This makes the codebase more adaptable to future requirements.
+
+By leveraging subcommands, the overall design becomes more robust, flexible, and easier to work with, effectively solving the identified challenges in the original implementation.
+
+
+
+
+== wallet-tool: Command
+
+Below shows the proposed format for the subcommand for WalletTool
+
+=== *`wallet-tool`*
+
+*Description*: Print and manipulate wallets. This command is the main command which requires that you specify a subcommand. You may provide a wallet file as an argument. Additionally, you can specify the network with an optional `--net` parameter. If `--net` is not specified, the network will be derived from the wallet.
+
+==== *Options*:
+
+* `--net ++<++NETWORK++>++`: Specifies the network.
+
+==== *Parameter*:
+* `++<++WALLET++_++FILE++>++`: Specifies what wallet file to load and save. If not specified, wallet will be created in default path with a default name. Only subcommand `create` allows you to exempt this parameter.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool <WALLET_FILE> [--net <NETWORK>] <Subcommand>
+----
+
+==== *Examples*:
+
+[arabic]
+. Specify a wallet file and subcommand with a specific network:
++
+[source,bash]
+----
+wallet-tool mywallet.wallet --net mainnet dump
+----
+
+. Create a new wallet with the `create` subcommand:
++
+[source,bash]
+----
+wallet-tool newwallet.wallet create
+----
+
+== Wallet-tool: Subcommands
+=== *`create`*
+
+*Description*: Creates a new wallet in the specified file. This command supports deterministic wallet seeds, watch-only wallets, and various configurations like timestamps and address derivation types. If `--seed` or `--watchkey` is combined with either `--date` or `--unixtime`, use that as a birthdate for the wallet.
+
+==== **Usage**:
+
+[source,bash]
+----
+wallet-tool <WALLET_FILE> create [--seed <SEED>] [--watchkey <XPUB>] [--date <DATE>] [--unixtime <UNIX_TIMESTAMP>] [--output-script-type <TYPE>] [--force]
+----
+
+==== **Options**:
+- `--seed <SEED>`: Specifies a mnemonic code or raw seed in hex/base58 raw seed bytes.
+- `--watchkey <XPUB>`: If present, it creates a watching wallet using the specified base58 xpub.
+- `--date <DATE>`: Wallet creation date (formatted as `YYYY-MM-DD`).
+- `--unixtime <UNIX_TIMESTAMP>`: Wallet creation time in Unix timestamp format (overrides `--date` if both are specified).
+- `--output-script-type <TYPE>`: If specified, use that for deriving addresses. Provide an output script type. Valid values: P2PKH, P2WPKH. Default: P2PKH.
+- `--force`: Overwrites any existing wallet file.
```

Comment:
Add --net

Reply:
I think we should default to `P2WPKH` it's the best choice we currently support.

Reply:
I'd like to add to this one: when doing or describing the migration, I think we should not change any behaviour at all, also not the defaults. Refactoring is refactoring, and changes are changes – separate commits, separate PRs (in most cases)

I but strongly agree the default should be `P2WPKH`. I'm a bit embarrassed why I missed this for such a long time.

So my proposal is we change the default now, before merging the migration.